src/System.Globalization/tests/CultureInfo/CultureInfoDateTimeFormat.cs

```diff
@@ -39,55 +39,14 @@ public void PosTest2()
         }
 
         [Theory]
-        [InlineData("en-US", "h:mm tt")]
-        public void TestShortTimePatternLocale(string locale, string expected)
+        [InlineData("fi-FI")]
+        public void TestLongTimePatternLocale2(string locale)
         {
             CultureInfo myTestCulture = new CultureInfo(locale);
-            DateTimeFormatInfo dtf = myTestCulture.DateTimeFormat;
-            Assert.Equal(expected, dtf.ShortTimePattern);
-        }
-
-        [Theory]
-        [InlineData("fi-FI", "H:mm", "H.mm")]
-        public void TestShortTimePatternLocale2(string locale, string expectedWindows, string expectedIcu)
-        {
-            CultureInfo myTestCulture = new CultureInfo(locale);
-            DateTimeFormatInfo dtf = myTestCulture.DateTimeFormat;
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                // Windows 10 uses ICU data here, this should be cleaned up as part of #3243
-                Assert.True(dtf.ShortTimePattern == expectedWindows || dtf.ShortTimePattern == expectedIcu);
-            }
-            else
-            {
-                Assert.Equal(expectedIcu, dtf.ShortTimePattern);
-            }
-        }
-
-        [Theory]
-        [InlineData("en-US", "h:mm:ss tt")]
-        public void TestLongTimePatternLocale(string locale, string expected)
-        {
-            CultureInfo myTestCulture = new CultureInfo(locale);
-            DateTimeFormatInfo dtf = myTestCulture.DateTimeFormat;
-            Assert.Equal(expected, dtf.LongTimePattern);
-        }
-
-        [Theory]
-        [InlineData("fi-FI", "H:mm:ss", "H.mm.ss")]
-        public void TestLongTimePatternLocale2(string locale, string expectedWindows, string expectedIcu)
-        {
-            CultureInfo myTestCulture = new CultureInfo(locale);
-            DateTimeFormatInfo dtf = myTestCulture.DateTimeFormat;
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                // Windows 10 uses ICU data here, this should be cleaned up as part of #3243
-                Assert.True(dtf.LongTimePattern == expectedWindows || dtf.LongTimePattern == expectedIcu);
-            }
-            else
-            {
-                Assert.Equal(expectedIcu, dtf.LongTimePattern);
-            }
+            DateTimeFormatInfo myDateTimeFormat = new DateTimeFormatInfo();
+            myDateTimeFormat.LongTimePattern = "H:mm:ss";
+            myTestCulture.DateTimeFormat = myDateTimeFormat;
+            Assert.Equal("H:mm:ss", myTestCulture.DateTimeFormat.LongTimePattern);
```

Comment:
previously we were literally testing the pattern itself from the culture data which is wrong because we cannot assume the data will never change. what I am testing now is explicitly setting a custom data (like H:mm:ss) and checking this format persist. 
We have other tests already testing the formatting and the parsing which does the round tripping checks for the date/time values